tough/src/editor/mod.rs

```diff
@@ -394,4 +621,127 @@ impl RepositoryEditor {
+        timestamp_expiration: DateTime<Utc>,
+        snapshot_expiration: DateTime<Utc>,
+    ) -> Result<SignedRepository>
+    where
+        T: Transport,
+    {
+        let rng = SystemRandom::new();
+        let signed_delegations =
+            SignedRole::<Targets>::new_targets(&repo.targets.signed, &[], &rng, true)?;
+        let signed_targets = SignedRole::from_signed(repo.targets)?;
+        let signed_root = SignedRole::from_signed(repo.root)?;
+
+        let mut snapshot = repo.snapshot.signed;
+        // Update snapshot version and expiration
+        snapshot.expires = snapshot_expiration;
+        snapshot.version = NonZeroU64::new(
+            u64::from(snapshot.version)
+                .checked_add(1)
+                .ok_or_else(|| error::Error::Overflow)?,
+        )
+        .ok_or_else(|| error::Error::Overflow)?;
+        // Snapshot stores metadata about targets and root
+        let targets_meta = Self::snapshot_meta(&signed_targets);
+        let root_meta = Self::snapshot_meta(&signed_root);
+        snapshot
+            .meta
+            .insert("targets.json".to_owned(), targets_meta);
+        snapshot.meta.insert("root.json".to_owned(), root_meta);
+
+        for (role, targets) in &signed_delegations {
+            let meta = Self::snapshot_meta(targets);
+            snapshot.meta.insert(format!("{}.json", role), meta);
+        }
+
+        let signed_snapshot = SignedRole::new(snapshot, &signed_root.signed.signed, keys, &rng)?;
+
+        // Update timestamp to reflect changes to snapshot
+        let mut timestamp = repo.timestamp.signed;
+        timestamp.expires = timestamp_expiration;
+        timestamp.version = NonZeroU64::new(
+            u64::from(timestamp.version)
+                .checked_add(1)
+                .ok_or_else(|| error::Error::Overflow)?,
+        )
+        .ok_or_else(|| error::Error::Overflow)?;
+        // Timestamp stores metadata about snapshot
+        let snapshot_meta = Self::timestamp_meta(&signed_snapshot);
+        timestamp
+            .meta
+            .insert("snapshot.json".to_owned(), snapshot_meta);
+
+        let signed_timestamp = SignedRole::new(timestamp, &signed_root.signed.signed, keys, &rng)?;
+
+        Ok(SignedRepository {
+            timestamp: signed_timestamp,
+            root: signed_root,
+            delegations: signed_delegations,
+            targets: signed_targets,
+            snapshot: signed_snapshot,
+        })
+    }
+
+    /// Crawls a given directory and symlinks any targets found to the given
+    /// "out" directory. If consistent snapshots are used, the target files
+    /// are prefixed with their `sha256`.
+    ///
+    /// For each file found in the `indir`, the method gets the filename and
+    /// if the filename exists in `Targets`, the file's sha256 is compared
+    /// against the data in `Targets`. If this data does not match, the
+    /// method will fail.
+    pub fn link_targets<P1, P2>(
```

Comment:
Did these get moved over from `SignedRepository`?  How come?  If they're needed here, let's make sure it's clear why you would use these rather than the methods from `SignedRepository`.